### PR TITLE
Signature of nested type with generic type parameter

### PR DESCRIPTION
**Pull request by nojaf**
When trying to create a signature for:

```fsharp
open System.Collections.Immutable

type ImmutableArrayViaBuilder<'T>(builder: ImmutableArray<'T>.Builder) = class end
```

the compiler generates:

```fsharp
type ImmutableArrayViaBuilder<'T> =
  new: builder: System.Collections.Immutable.ImmutableArray`1.Builder<'T> ->
         ImmutableArrayViaBuilder<'T>
```

``System.Collections.Immutable.ImmutableArray`1.Builder<'T>`` is the problematic bit, 
it should be ``System.Collections.Immutable.ImmutableArray<'T>.Builder``.

I'm guessing [Builder](https://learn.microsoft.com/en-us/dotnet/api/system.collections.immutable.immutablearray-1.builder?view=net-7.0) is a nested class?

My initial fix is not the way to go. And I could use a tip on how to tackle this in a nicer way.
I'm all hears 😇.

PS: @T-Gro I think my test will only work on net7.0, any pointers on how to address that?

### CODE REVIEW

**Comment by vzarytovskii** [comment]
> PS: @T-Gro I think my test will only work on net7.0, any pointers on how to address that?

`#ifdef` it, but will probably need to do the `_OR_GREATER` veresion. Or ignore it for non-net7

**Comment by nojaf** [comment]
A slightly better attempt by storing the generic parameters in `ModuleOrType`.
I'm most likely still missing some edge cases.

File: tests/FSharp.Compiler.ComponentTests/Signatures/NestedTypeTests.fs [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+"""
+    |> withReferences [ CSLib ]
+    |> printSignatures
+    |> should equal
+        """
+module Sample
+
+val f: g: Lib.Root<System.TimeSpan,System.TimeSpan,System.TimeSpan,System.TimeSpan,System.TimeSpan>.Foo<int,float,string,System.DateTime>.Bar<char,int,string> -> unit"""
+
+#if NETCOREAPP
```
Comment (T-Gro) [inline tests/FSharp.Compiler.ComponentTests/Signatures/NestedTypeTests.fs:82]: You can use `[<FactForNETCOREAPP>]`, it was created for this purpose.

(it will also allow us to see such tests at a glance once we want to have an overview of platform-specific features)

Comment (nojaf) [reply tests/FSharp.Compiler.ComponentTests/Signatures/NestedTypeTests.fs]: Thanks

**Comment by nojaf** [comment]
This is ready.

File: tests/FSharp.Compiler.ComponentTests/Signatures/NestedTypeTests.fs [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+[<Fact>]
+let ``ImmutableArray<'T>.Builder roundtrip`` () =
```
[the remark this answers is not in the document]
Comment (T-Gro) [reply tests/FSharp.Compiler.ComponentTests/Signatures/NestedTypeTests.fs:84]: I think you should be able to have #if NETCOREAPP inside of that test file (and e.g. just an empty module in the desktopframework case)

[the remark this answers is not in the document]
Comment (nojaf) [reply tests/FSharp.Compiler.ComponentTests/Signatures/NestedTypeTests.fs:84]: image
That doesn't work, there is no `--define` pass by default in https://github.com/dotnet/fsharp/blob/b456cecba869065ab8488e53583598dd672c61a3/tests/FSharp.Test.Utilities/CompilerAssert.fs#L313-L337

**Review by edgarfgp — APPROVED** [review]

**Review by T-Gro — APPROVED** [review]

**Review by psfinaki — APPROVED** [review]
